flow/project.py

```diff
@@ -404,41 +273,43 @@ class FlowOperation(object):
         :class:`dict`
     """
 
-    def __init__(self, cmd, pre=None, post=None, directives=None):
+    def __init__(self, cmd, pre=None, post=None, directives=None, aggregate=False):
         if pre is None:
             pre = []
         if post is None:
             post = []
         self._cmd = cmd
         self.directives = directives
+        self.aggregate = aggregate
 
         self._prereqs = [FlowCondition(cond) for cond in pre]
         self._postconds = [FlowCondition(cond) for cond in post]
 
     def __str__(self):
         return "{type}(cmd='{cmd}')".format(type=type(self).__name__, cmd=self._cmd)
 
-    def eligible(self, job):
+    def eligible(self, *jobs):
         "Eligible, when all pre-conditions are true and at least one post-condition is false."
-        pre = all(cond(job) for cond in self._prereqs)
+        pre = all(cond(*jobs) for cond in self._prereqs)
         if pre and len(self._postconds):
-            post = any(not cond(job) for cond in self._postconds)
+            post = any(not cond(*jobs) for cond in self._postconds)
         else:
             post = True
         return pre and post
 
-    def complete(self, job):
+    def complete(self, *jobs):
         "True when all post-conditions are met."
         if len(self._postconds):
-            return all(cond(job) for cond in self._postconds)
+            return all(cond(job) for cond in self._postconds for job in jobs)
```

Comment:
This is inconsistent with above, where we have `cond(*jobs)`, right? Are we assuming that conditions are now functions of all jobs, or just one job? From our discussion, my understanding was that we would leave them as functions of one job but then add a new way to specify aggregate conditions.